repository_service_tuf_api/targets.py

```diff
@@ -108,25 +108,25 @@ def post(payload: AddPayload) -> Response:
         )
 
     task_id = get_task_id()
-    if payload.add_task_id_to_custom:
+    if payload.add_task_id_to_custom is True:
         new_targets: List[Targets] = []
         for target in payload.targets:
-            if target.info.custom is None:
-                target.info.custom = {}
-
-            # Add task_id info in custom while keeping the old custom
-            target.info.custom = {
-                "added_by_task_id": task_id,
-                **target.info.custom,
-            }
+            if target.info.custom:
+                target.info.custom = {
+                    "added_by_task_id": task_id,
+                    **target.info.custom,
+                }
```

Comment:
I just realized that nowhere it's specified that `custom` should be a dictionary.
The TUF specification defines `custom` [like this](https://theupdateframework.github.io/specification/latest/#custom):
```
An object. If defined, the elements and values of the CUSTOM object will be made
available to the client application. The format of the CUSTOM object is opaque to the
framework, which only needs to know that the "custom" attribute maps to an object.
The CUSTOM object may include version numbers, dependencies, requirements, or
any other data that the application wants to include to describe the file at TARGETPATH.
The application may use this information to guide download decisions.
```
In `python-tuf` when we access the `custom` field we don't say that the return type is a `Dict` or a `Map`:
https://github.com/theupdateframework/python-tuf/blob/a6460c60ba9ed3a609e43ab981ec5a941606d6f4/tuf/api/metadata.py#L1776

Maybe we should brainstorm what should we do if there is an existing `custom` which is not a `Dict` or a `Map`?

Reply:
We can open an issue if you agree there is a sense to discuss this.

Reply:
The RSTUF defines some constraints in the Metadata, such as the delegations to succinct hash bin roles. The main reason is to make TUF easily adoptable and give some flexibility for use cases.
In that sense, using custom as a `Dict` helps the current TUF users that want to migrate to RSTUF once they need to define a key value for their specific custom value type.